File: cases/taint/python/closure-capture-positive/closure_capture.py
def dfb_source():  # DFB-SOURCE: closure-capture-input
    return "tainted"


def dfb_sink(value):  # DFB-SINK: closure-capture-sink
    pass


def make_reporter():
    captured = dfb_source()  # DFB-WITNESS: closure-capture-bind

    def reporter():
        dfb_sink(captured)

    return reporter


def run():
    reporter = make_reporter()
    reporter()

File: cases/taint/ruby/closure-capture-positive/closure_capture.rb
def dfb_source # DFB-SOURCE: closure-capture-input
  "tainted"
end

def dfb_sink(value) # DFB-SINK: closure-capture-sink
end

def make_reporter
  captured = dfb_source # DFB-WITNESS: closure-capture-bind
  -> { dfb_sink(captured) }
end

def run
  reporter = make_reporter
  reporter.call
end
